--- ./ltdl.c	2003-10-12 12:45:30.000000000 -0500
+++ raw-ltdl.guilemod.c	2003-10-12 13:13:34.000000000 -0500
@@ -1,5 +1,5 @@
 /* ltdl.c -- system independent dlopen wrapper
-   Copyright (C) 1998, 1999, 2000 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002 Free Software Foundation, Inc.
    Originally by Thomas Tanner <tanner@ffii.org>
    This file is part of GNU Libtool.
 
@@ -26,7 +26,7 @@
 */
 
 #if HAVE_CONFIG_H
-#  include <config.h>
+#  include "config.h"
 #endif
 
 #if HAVE_UNISTD_H
@@ -124,13 +124,15 @@
 #  define assert(arg)	((void) 0)
 #endif
 
-#include "ltdl.h"
+#include "raw-ltdl.h"
 
 #if WITH_DMALLOC
 #  include <dmalloc.h>
 #endif
 
 
+/* Saves on those hard to debug '\0' typos....  */
+#define LT_EOS_CHAR	'\0'
 
 
 /* --- WINDOWS SUPPORT --- */
@@ -387,7 +389,7 @@
 
   for (i = 0; i < size; ++i)
     {
-      dest[i] = src[i];
+      ((char *) dest)[i] = ((char *) src)[i];
     }
 
   return dest;
